### PR TITLE
RK-12390 - Initial version of the java tutorial

### PR DESCRIPTION
**Pull request by LiranLast**

### CODE REVIEW

File: docs/java-container-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Configuration is where you can get fancy. You have got additional options up your sleeve:
+1. Move options from environment variables to secret or configuration managers.
+2. If you are using a [Rookout Controller](etl-controller-intro), set up the remote host and port configuration.
+3. Dig deeper into other options available right [here](jvm-setup#sdk-configuration).
+
+### 3. Embed Source Information
+
+Rookout offers the smoothest debugging experience by fetching up-to-date source code for each server.
+
+Set this up for containerized applications by adding a handful of files from your `.git` folder to the container image.
```
Comment (ElDuderinos) [inline docs/java-container-tutorial.md:58]: Maybe add a clarification about how this doesn't add your source code into the image, only some metadata ?

File: docs/java-container-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+!.git/HEAD
+!.git/config
+!.git/refs
+```
+
+Add a final `copy` command to the `Dockerfile`.
+```docker
+COPY .git /.git
+```
+
```
Comment (ElDuderinos) [inline docs/java-container-tutorial.md:74]: Add reference to here - https://www.rookout.com/blog/embedding-source-code-version-information-in-docker-images/ ?
With "Need to understand what did we just do?" and a link

File: docs/java-container-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+### 1. Add the Java Agent
+
+Rookout for Java (and any other JVM-based language) is a simple Java Agent.  
+Deploy it by downloading the JAR to the image and instructing the JVM to load it.
+
+Open the `Dockerfile` and add the following two lines just above the `ENTRYPOINT` command:
+```docker
+RUN curl -L "https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.rookout&a=rook&v=LATEST" -o rook.jar
+ENV JAVA_TOOL_OPTIONS "-javaagent:rook.jar"
```
Comment (ElDuderinos) [inline docs/java-container-tutorial.md:30]: This makes some assumptions regarding the ENTRYPOINT and the application's JAR file's path.

File: docs/java-container-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+---
+
+This short tutorial will walk you through the perfect Rookout deployment for containerized Java applications in three quick steps.
+
+For any follow-up questions, check out the full docs for the Rookout [Java Agent](jvm-setup) or reach out!
+
+### Get Your Application
+
+First things first, choose an application.
+If you don't have one readily available, use our [sample application](https://github.com/Rookout/java-tutorial-2022).
```
Comment (ElDuderinos) [inline docs/java-container-tutorial.md:14]: Let's use the power of Github and add a branch/tag in that repository with Rookout integrated.
That way you'll be able to add a link to the final docker file, or even a simple link that show's the difference between the new dockerfile and the old one.
Even simple things like your instructions on adding a token (ENV ROOKOUT_TOKEN [Your Rookout Token]) can be confusing without an example ('hey, do I need double quotes? do I need to keep those square brackets?').

Comment (LiranLast) [reply docs/java-container-tutorial.md:14]: We'll do once the first version is finalized :)

File: docs/java-container-tutorial.md (hunk 2 above)
Comment (LiranLast) [reply docs/java-container-tutorial.md:74]: Added a link :)

File: docs/java-container-tutorial.md (hunk 1 above)
Comment (LiranLast) [reply docs/java-container-tutorial.md]: Trying to keep this short - added a link that will explain :)

**Pull request merged**